application/api/dos_interface/reporting.py

```diff
@@ -0,0 +1,99 @@
+from django.core.exceptions import ObjectDoesNotExist
+from api.dos_interface.queries import get_service_info
+
+from datetime import datetime
+from time import gmtime, strftime
+
+import logging
+
+usage_reporting_logger = logging.getLogger("api.usage.reporting")
+
+
+def _get_service_info(service_uid):
+    try:
+        service_info, parent_service_info, dos_region_info = get_service_info(service_uid)
+        return service_info, parent_service_info, dos_region_info
+    except ObjectDoesNotExist:
+        return None
+
+
+def _retrieve_service_data(service_uid):
+
+    service_info, parent_service_info, dos_region_info = _get_service_info(service_uid)
+
+    service_data = {"CLIENT_NAME": service_info["modifiedby"]}
+    service_data["ORG_NAME"] = "org_name=" + service_info["name"]
+    service_data["ORG_TYPE"] = "org_type=" + str(service_info["typeid"])
+
+    parent_org = ""
+    if parent_service_info is not None:
+      parent_org = parent_service_info["uid"]
```

Comment:
Might worth checking with Jayne if the DoS Region is considered the parent when `parent_service_info` is `None` here. As in DoS the DoS Region (service) would technically be the parent.

Reply:
Good comment. Yes, the region will be the parent. And in this case the parent_service_info dictionary will contain the same as the dos_region_info dict.  The only way for the parent_service_info dict to be None is if we are changing the state of the top level DoS Region, and when that's the case, there is no parent.  So, the parent will either be another service or it will be the DoS Region itself (which will be both the second row in the list AND the last). Does that make sense?

Reply:
Yeah, that makes sense 👍